### PR TITLE
feat(replay): Replay Web Vital Breadcrumbs

### PR DESCRIPTION
**Pull request by c298lee**
Enables replay web vital breadcrumbs if feature flag is on.
Example:
image

Relates to https://github.com/getsentry/sentry/issues/69881

### CODE REVIEW

File: static/app/utils/replays/getFrameDetails.tsx [hunk 1]
```diff
@@ -274,24 +276,34 @@ const MAPPER_FOR_FRAME: Record<string, (frame) => Details> = {
...
-    icon: <IconInfo size="xs" />,
-  }),
+  'web-vital': (frame: WebVitalFrame) => {
+    switch (frame.data.rating) {
+      case 'good':
+        return {
+          color: 'green300',
+          description: `Good ${Math.round(frame.data.value * 100) / 100}ms`,
+          tabKey: TabKey.NETWORK,
+          title: frame.description.replaceAll('-', ' '),
```
Comment (ryan953) [inline static/app/utils/replays/getFrameDetails.tsx:286]: Might be able to call `toTitleCase(explodeSlug(frame.description))` for these titles

File: static/app/utils/replays/getFrameDetails.tsx [hunk 2]
```diff
@@ -274,24 +276,34 @@ const MAPPER_FOR_FRAME: Record<string, (frame) => Details> = {
...
-    tabKey: TabKey.NETWORK,
-    title: 'LCP',
-    icon: <IconInfo size="xs" />,
-  }),
+  'web-vital': (frame: WebVitalFrame) => {
+    switch (frame.data.rating) {
+      case 'good':
+        return {
+          color: 'green300',
+          description: `Good ${Math.round(frame.data.value * 100) / 100}ms`,
```
Comment (ryan953) [inline static/app/utils/replays/getFrameDetails.tsx:284]: looks like we're trying to limit it to 2 decimal places?

```
description: `Good ${frame.data.value.toFixed(2)}ms`,
```

File: static/app/utils/replays/replayDataUtils.tsx [hunk 3]
```diff
@@ -72,7 +72,7 @@ export function replayTimestamps(
     .map(rawCrumb => rawCrumb.timestamp)
     .filter(Boolean);
   const rawSpanDataFiltered = rawSpanData.filter(
-    ({op}) => op !== 'largest-contentful-paint'
+    ({op}) => op !== 'web-vital' && op !== 'largest-contentful-paint'
```
Comment (michellewzhang) [inline static/app/utils/replays/replayDataUtils.tsx:75]:
```suggestion
    ({op}) => op !== 'web-vital'
```

looks like we got rid of `largest-contentful-paint` op in static/app/utils/replays/types.tsx, so could we remove the second condition?

File: static/app/utils/replays/getFrameDetails.tsx [hunk 4]
```diff
@@ -274,24 +278,34 @@ const MAPPER_FOR_FRAME: Record<string, (frame) => Details> = {
...
-    tabKey: TabKey.NETWORK,
-    title: 'LCP',
-    icon: <IconInfo size="xs" />,
-  }),
+  'web-vital': (frame: WebVitalFrame) => {
+    switch (frame.data.rating) {
+      case 'good':
+        return {
+          color: 'green300',
+          description: `Good ${frame.data.value.toFixed(2)}ms`,
```
Comment (michellewzhang) [inline static/app/utils/replays/getFrameDetails.tsx:286]: could we translate the descriptions?

File: static/app/utils/replays/replayReader.tsx [hunk 5]
```diff
@@ -244,6 +264,7 @@ export default class ReplayReader {
   private _cacheKey: string;
   private _duration: Duration = duration(0);
   private _errors: ErrorFrame[] = [];
+  private _featureFlags: string[] | undefined = [];
```
Comment (ryan953) [inline static/app/utils/replays/replayReader.tsx:267]:
```suggestion
  private _featureFlags: string[] = [];
```

**Review by ryan953 — APPROVED** [review]

**Review by michellewzhang — APPROVED** [review]

**Pull request merged**